codegenerator/cli/templates/static/codegen/src/Config.res

```diff
@@ -120,6 +120,13 @@ let shouldRollbackOnReorg = config =>
   | _ => false
   }
 
+let shouldSaveHistory = (config, ~isInReorgThreshold) =>
+  switch config.historyConfig {
+  | {rollbackFlag: RollbackOnReorg} if isInReorgThreshold => true
+  | {historyFlag: FullHistory} => true
+  | _ => false
+  }
+
```

Comment:
Previously was using the shouldRollbackOnReorg function. But this is more specific. If we are saving full history it should always be true, if it's just standard rollback on reorg config it should only save if the indexer is in the reorg threshold